### PR TITLE
Desmakers 3492 xmc4ino update xmclib to be in sync with mtb

### PR DESCRIPTION
**Pull request by LinjingZhang**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

CONTRIBUTING.md also tells you what to expect in the PR process.

Description
update XMCLib to 4.3.0

Related Issue
/

Context
update cores\xmc_lib\XMCLib
- CAN macro manually changed 
- only XMC1400 device header file updated

### CODE REVIEW

**Comment by LinjingZhang** [comment]
https://github.com/LinjingZhang/XMC-for-Arduino/releases/tag/V4.0.0

I made a fork release, hope it will make the review/ quick test easier...  I know it could be hard to review this change

**Review by ederjc — APPROVED** [review]
The fork release made it way easier for me to test, thanks.
This is not a detailed review:
- I looked through the docs changes and they make sense
- I compile-checked for all platforms and tested with the CO2 sensor on XMC1400 Arduino Kit, works so far
- **Approved from what I could test now, but as discussed this needs further testing before merging in master**

File: .github/scripts/release.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 
 import argparse, copy, hashlib, json, re, requests, os, shutil
 
-version = '0.2.0'
```
Comment (jaenrig-ifx) [inline .github/scripts/release.py:4]: This release.py needs review together with the diff that we have now in master vs develop. By the way, we should merge the master into the develop after the last PR.

**Review by jaenrig-ifx — APPROVED** [review]
Just checked to build for some examples :) Looks good 👍 @LinjingZhang !
As Julian said, on target manual validation is necessary covering as much as reasonable possible the matrix of boards, built-in libraries and reference language API.

File: .github/scripts/release.py (hunk 1 above)
Comment (LinjingZhang) [reply .github/scripts/release.py:4]: Sure. We could review it together whenever you have time😄

Comment (jaenrig-ifx) [reply .github/scripts/release.py:4]: Yep, next week we can do it!

**Comment by LinjingZhang** [comment]
As discussed, I will do the test next sprint with students. So merge here for now~ 
btw, since the merge is to feature branch, I was merge the develop to feature/branch to keep it up to date. The diff include some overlapping changes you already reviewed...😉

**Pull request merged**